unitTests/utils/DependsOnManagerTest.ts

```diff
@@ -170,118 +242,123 @@ export function DependsOnManagerTest() {
+          triggerStateChangeOnDependentFacet();
+          expect(masterFacet.cmp.keepDisplayedValuesNextTime).toBeUndefined();
         });
 
-        it('the dependent facet is hidden at startup', () => {
-          expect($$(test.cmp.element).isVisible()).toBeFalsy();
+        it(`when the master facet id is unique,
+          it update keepDisplayedValuesNextTime`, () => {
+          masterFacet.cmp.keepDisplayedValuesNextTime = undefined;
+          expect(masterFacet.cmp.keepDisplayedValuesNextTime).toBeUndefined();
+          triggerStateChangeOnDependentFacet();
+          expect(masterFacet.cmp.keepDisplayedValuesNextTime).toBe(false);
         });
 
-        it('the master facet is visible at startup', () => {
-          expect($$(masterFacet.cmp.element).isVisible()).toBeTruthy();
+        it(`when resetting the master facet,
+        it resets the dependent facet`, () => {
+          masterFacet.cmp.reset();
+          triggerStateChangeOnDependentFacet();
+
+          expect(test.cmp.getSelectedValues().length).toBe(0);
+          expect(test.cmp.getExcludedValues().length).toBe(0);
         });
 
-        it(`when the master facet has one selected value,
-      the dependent facet is visible`, () => {
-          masterFacet.cmp.selectValue('master value');
-          Simulate.query(test.env, { results: getMasterAndDependentFacetResults() });
+        it(`when deselecting one of the master facet selected options,
+        it does not reset the dependent facet`, () => {
+          masterFacet.cmp.deselectValue(selectedMasterValue1);
+          triggerStateChangeOnDependentFacet();
 
-          expect($$(test.cmp.element).isVisible()).toBeTruthy();
+          expect(test.cmp.getSelectedValues().length).toBe(1);
+          expect(test.cmp.getExcludedValues().length).toBe(1);
         });
 
-        it(`when the master facet has one excluded value,
-      the dependent facet is still hidden`, () => {
-          masterFacet.cmp.excludeValue('excluded master value');
-          Simulate.query(test.env, { results: getMasterAndDependentFacetResults() });
+        it(`when deselecting both master facet selected options,
+        it resets the dependent facet`, () => {
+          masterFacet.cmp.deselectValue(selectedMasterValue1);
+          masterFacet.cmp.deselectValue(selectedMasterValue2);
+          triggerStateChangeOnDependentFacet();
 
-          expect($$(test.cmp.element).isVisible()).toBeFalsy();
+          expect(test.cmp.getSelectedValues().length).toBe(0);
+          expect(test.cmp.getExcludedValues().length).toBe(0);
         });
+      });
+    });
 
-        describe(`given the master facet has two selected values and the dependent facet has one selected and one excluded value`, () => {
-          const selectedMasterValue1 = 'selected master value 1';
-          const selectedMasterValue2 = 'selected master value 2';
+    describe('using DynamicFacet', () => {
```

Comment:
Tests start here

Reply:
I think for this PR it would be best to just remove the new tests in this file, and have the rework done in a separate PR.